internal/services/cosmos/cosmosdb_cassandra_datacenter_resource.go

```diff
@@ -195,63 +191,65 @@ func resourceCassandraDatacenterRead(d *pluginsdk.ResourceData, meta interface{}
-			d.Set("sku_name", props.Sku)
-			d.Set("availability_zones_enabled", props.AvailabilityZone)
+	if model := resp.Model; model != nil {
+		if props := model.Properties; props != nil {
+			if res := props; res != nil {
+				d.Set("delegated_management_subnet_id", props.DelegatedSubnetId)
+				d.Set("location", location.NormalizeNilable(props.DataCenterLocation))
+				d.Set("backup_storage_customer_key_uri", props.BackupStorageCustomerKeyUri)
+				d.Set("base64_encoded_yaml_fragment", props.Base64EncodedCassandraYamlFragment)
+				d.Set("managed_disk_customer_key_uri", props.ManagedDiskCustomerKeyUri)
+				d.Set("node_count", props.NodeCount)
+				d.Set("disk_count", int(*props.DiskCapacity))
+				d.Set("disk_sku", props.DiskSku)
+				d.Set("sku_name", props.Sku)
+				d.Set("availability_zones_enabled", props.AvailabilityZone)
+			}
 		}
 	}
 	return nil
 }
 
 func resourceCassandraDatacenterUpdate(d *pluginsdk.ResourceData, meta interface{}) error {
-	client := meta.(*clients.Client).Cosmos.CassandraDatacentersClient
+	client := meta.(*clients.Client).Cosmos.ManagedCassandraClient
 	ctx, cancel := timeouts.ForUpdate(meta.(*clients.Client).StopContext, d)
 	defer cancel()
 
-	id, err := parse.CassandraDatacenterID(d.Id())
+	id, err := managedcassandras.ParseDataCenterID(d.Id())
 	if err != nil {
 		return err
 	}
 
-	body := documentdb.DataCenterResource{
-		Properties: &documentdb.DataCenterResourceProperties{
-			DelegatedSubnetID:  utils.String(d.Get("delegated_management_subnet_id").(string)),
-			NodeCount:          utils.Int32(int32(d.Get("node_count").(int))),
+	body := managedcassandras.DataCenterResource{
+		Properties: &managedcassandras.DataCenterResourceProperties{
+			DelegatedSubnetId:  utils.String(d.Get("delegated_management_subnet_id").(string)),
+			NodeCount:          utils.Int64(int64(d.Get("node_count").(int))),
 			DataCenterLocation: utils.String(azure.NormalizeLocation(d.Get("location").(string))),
 			DiskSku:            utils.String(d.Get("disk_sku").(string)),
 		},
 	}
 
 	if v, ok := d.GetOk("backup_storage_customer_key_uri"); ok {
-		body.Properties.BackupStorageCustomerKeyURI = utils.String(v.(string))
+		body.Properties.BackupStorageCustomerKeyUri = utils.String(v.(string))
 	}
 
 	if v, ok := d.GetOk("base64_encoded_yaml_fragment"); ok {
 		body.Properties.Base64EncodedCassandraYamlFragment = utils.String(v.(string))
 	}
 
 	if v, ok := d.GetOk("managed_disk_customer_key_uri"); ok {
-		body.Properties.ManagedDiskCustomerKeyURI = utils.String(v.(string))
+		body.Properties.ManagedDiskCustomerKeyUri = utils.String(v.(string))
 	}
 
-	future, err := client.CreateUpdate(ctx, id.ResourceGroup, id.CassandraClusterName, id.DataCenterName, body)
+	future, err := client.CassandraDataCentersCreateUpdate(ctx, *id, body)
 	if err != nil {
 		return fmt.Errorf("updating %q: %+v", id, err)
 	}
 
-	if err = future.WaitForCompletionRef(ctx, client.Client); err != nil {
+	if err = future.Poller.PollUntilDone(); err != nil {
```

Comment:
we can use `CassandraDataCentersCreateUpdateThenPoll` here?

Reply:
Updated